File: POO/2/2.java
public class Bus {
    private int capacidad;
    private int pasajeros;
    private double recaudacion;
    public Bus(int capacidad){
        this.capacidad = capacidad;
        this.pasajeros = 0;
        this.recaudacion = 0.0;
    }
    public int subir(int cantidad){
        int disponibles = capacidad - pasajeros;
        int subir = cantidad <= disponibles ? cantidad : disponibles;
        pasajeros += subir;
        return subir;
    }
    public double cobrarPasaje(){
        double costo = 1.5;
        double total = pasajeros * costo;
        recaudacion += total;
        return total;
    }
    public int asientosDisponibles(){
        return capacidad - pasajeros;
    }
    public static void main(String[] args){
        Bus b = new Bus(40);
        System.out.println(b.subir(25));
        System.out.println(b.cobrarPasaje());
        System.out.println(b.asientosDisponibles());
    }
}

File: POO/2/2.py
class Bus:
    def __init__(self, capacidad):
        self.capacidad = capacidad
        self.pasajeros = 0
        self.recaudacion = 0.0
    def subir(self, cantidad):
        disponibles = self.capacidad - self.pasajeros
        subir = cantidad if cantidad <= disponibles else disponibles
        self.pasajeros += subir
        return subir
    def cobrar_pasaje(self):
        costo = 1.5
        total = self.pasajeros * costo
        self.recaudacion += total
        return total
    def asientos_disponibles(self):
        return self.capacidad - self.pasajeros

if __name__ == "__main__":
    b = Bus(40)
    print(b.subir(25))
    print(b.cobrar_pasaje())
    print(b.asientos_disponibles())
